h2o-algos/src/main/java/hex/gam/MatrixFrameUtils/GenerateGamMatrixOneColumn.java

```diff
@@ -48,19 +48,29 @@ public void map(Chunk[] chk, NewChunk[] newGamCols) {
     for (int rowIndex=0; rowIndex < chunkRows; rowIndex++) {
       double gamRowSum = 0.0;
       // find index of knot bin where row value belongs to
-      double xval = chk[0].atd(rowIndex);
-      int binIndex = locateBin(xval,_knots); // location to update
-      // update from F matrix F matrix = [0;invB*D;0] and c functions
-      GamUtilsCubicRegression.updateFMatrixCFunc(basisVals, xval, binIndex, _knots, crSplines._hj, _bInvD);
-      // update from a+ and a- functions
-      GamUtilsCubicRegression.updateAFunc(basisVals, xval, binIndex, _knots, crSplines._hj);
-      // copy updates to the newChunk row
-      for (int colIndex = 0; colIndex < _numKnots; colIndex++) {
-        newGamCols[colIndex].addNum(basisVals[colIndex]);
-        gamRowSum += Math.abs(basisVals[colIndex]);
+      if (chk[1].atd(rowIndex) != 0) {  // consider weight column value during gamification.  If 0, insert rows of NaN.
```

Comment:
The comment says row this row of the matrix will be inserted with `NaN`, yet in the `else` block `0`s are inserted. I believe zeros are correct, as when substituted to the final linear formula <img src="https://render.githubusercontent.com/render/math?math=Y_{i}=\sum_{j=1}^{k}b_{j}(x_{i})\beta_{j}+\beta_{0}">, the result will be zero, as the result of the polynomial function(s) approximating the space between knots will be zero.

Am I right @wendycwong ?

Reply:
Did we address this @wendycwong ? The comment still says "If 0, insert rows of NaN" - but that's not what is happening in the `else` block. See comment above.

Reply:
@pavel:  I am such a dumb ass.  I fixed it in my branch without pushing it.  I just pushed and it looks correct now.  Thanks for catching this.

Reply:
Oh my god, no you're not.

Thank you @wendycwong .